### DISCUSSION TITLE
Duplicating a repository through github's interface

### DISCUSSION THREAD

**Discussion by carreraSilvio** [community/community#123235] [Repositories] [answered]
### Select Topic Area

Question

### Body

Is there a way to duplicate a repo on github through its interface? I have a repo I use for prototypes, and once the prototype has matured I want to have it in its own repo. 

For my case it would just be easier to duplicate the original repo and delete and what I don't want afterwards. I know that I can just clone the repo, and push it to the new one, but I was wondering if there was a simple way through the UI.

**Comment by MostlyKIGuess** [comment, marked as answer]
No direct duplicate function on GitHub's interface, but two options:

1. **Forking (Faster):**
   - Go to the original repository.
   - Click "Fork" button (top right).
   - Creates a copy under your account with history.
   - Rename & remove unwanted files in your fork.

2. **Clone & Push (More Control):**
   - Get original repo's HTTPS clone URL (Code button).
   - Create a new repository on your GitHub account.
   - Clone it locally using `git clone <URL>`.
   - Navigate to the local directory (`cd`).
   - Initialize Git (`git init` if needed).
   - Add all files (`git add .`).
   - Commit changes (`git commit -m "Initial commit"`).
   - Push to your new repo (`git push origin main`).

**Reply by zackarydev** [reply]
There's a bug where we can't create if we don't change user/owner !

**Comment by krishealty** [comment]
well you can do it by making an **organisation**

my organisation>create new organisation>add new repository>import repository

i think this will duplicate the repository with the interface.

**Comment by aditi5050** [comment]
While GitHub doesn't have a literal "Duplicate" button for standard repositories, you are completely right that doing this through the UI is much easier than doing the whole clone/change-remote/push dance in the terminal.

Depending on whether you want to keep your prototype's commit history or start with a clean slate, there are two simple ways to duplicate a repository entirely through the GitHub web interface.

### Method 1: The "Import" Tool (Keeps Commit History)

This is the closest thing to a true duplication because it clones the exact state of the repository, including every single commit you made while building the prototype.

1. Click the **`+`** icon in the very top right corner of the GitHub navigation bar and select **Import repository** (or go directly to `https://github.com/new/import`).
2. In the "Your old repository's clone URL" field, simply paste the web URL of your original prototype repository.
3. Choose your account as the owner and type in the name for your new repository.
4. Select the visibility (Public or Private) and click **Begin import**.

*Note: If your prototype repository is private, GitHub will prompt you for your credentials. You will need to use a Personal Access Token (PAT) as the password to authorize the import.*

### Method 2: The "Template" Feature (Clean Commit History)

If your prototype phase involved a lot of messy, trial-and-error commits that you don't want to carry over to the mature project, you can use the template feature. This copies all your files and directory structures exactly as they are, but gives you a brand new, clean commit history.

1. Navigate to your original prototype repository and click on the **Settings** tab.
2. In the main "General" section, check the box labeled **Template repository**.
3. Go back to the main code tab of that repository.
4. You will now see a green **Use this template** button near your code. Click it, and select **Create a new repository**.
5. Fill out the name and details for your new repository and click **Create repository from template**.

Once your repo is duplicated using either method, you can jump right into the new repository's UI, delete the files or folders you don't need for the mature version, and you're good to go! Once you are done, you can also uncheck the "Template repository" box in your original repo if you used the second method.
